lib/setupmanagers/qemuhck/devices/iommu.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+module AutoHCK
+  class QemuHCK
+    module Devices
+      extend T::Sig
+
+      sig { params(logger: AutoHCK::MultiLogger).returns(Models::QemuHCKDevice) }
+      def self.iommu(logger)
+        cpuinfo = File.read('/proc/cpuinfo')
+        vendor_match = cpuinfo.match(/^vendor_id\s*:\s*(\S+)/)
+
+        if vendor_match.nil?
+          logger.fatal('Could not determine CPU vendor')
+          exit 1
```

Comment:
Please `raise` instead of `exit`. It's the caller to decide how to handle an error.